tests/SemanticAnalysis.unit_test.cpp

```
@@ -0,0 +1,199 @@
+				DiagTypeToString(diag.diagType), DiagSeverityToString(diag.severity),
+				(diag.faultyNodeID.has_value()) ? std::to_string(diag.faultyNodeID.value()) : "",
+				(diag.faultyNodePin.has_value()) ? std::format(", Pin \"{}\"", diag.faultyNodePin.value()) : "",
+				diag.message
+			);
+
+		std::cout << '\n';
+	}
+}
+
+
+TEST_CASE("Semantic Analysis: Simple Graph", __FILE__) {
```

Cppcheck (2.21.0)

[style] The function 'C_A_T_C_H_T_E_S_T_3' is never used.

(unusedFunction)